cogs/birthday/birthday.py

```diff
@@ -416,6 +417,178 @@ async def deleteMemberBirthday(self, ctx: Context, member: discord.Member):
+                        f"{italics('(For your privacy, this message will disappear shortly.)')}"
+                    )
+                    sentReplyMsg: discord.Message = await ctx.send(replyMsg)
+                    await asyncio.sleep(5)
+                    await sentReplyMsg.delete()
+                    return
+
+        setSelfBirthdayCmd: commands.Command = self.setSelfBirthday
+        setSelfBirthdayCmdStr = (
+            # pylint: disable=no-member
+            f"`{ctx.clean_prefix}help {setSelfBirthdayCmd.qualified_name}`"
+        )
+
+        replyMsg = (
+            f"{headerBad}\n"
+            "Your birthday in this server has not been set. "
+            "Please contact an administrator/moderator, or, "
+            "if it is allowed by the server's admins and/or "
+            "moderators, try setting it yourself. Reference "
+            f"{setSelfBirthdayCmdStr} for the command syntax."
+        )
+        await ctx.send(replyMsg)
+
+    @me.command("set", aliases=["add"])
+    @commands.guild_only()
+    async def setSelfBirthday(self, ctx: Context, month: int, day: int):
+        """Set your birthday.
+
+        Specify your birth month and birth day in numbers.
+
+        Parameters
+        ----------
+        month: int
+            Birth month.
+        day: int
+            Birth day.
+        """
+        fnTitle = "Birthday - Set Self's Birthday"
+        headerBad = f":negative_squared_cross_mark: {bold(fnTitle)}"
+        headerGood = f":white_check_mark: {bold(fnTitle)}"
+
+        if not await self.config.guild(ctx.guild).get_attr(KEY_ALLOW_SELF_BDAY)():
+            await ctx.send(
+                f"{headerBad}\n"
+                "This function is not enabled. You cannot set your birthday. "
+                "Please let an administrator or a moderator know if you "
+                "believe this function should be enabled."
+            )
+            return
+
+        birthdayConfig = self.config.member(ctx.author)
+        if birthdayConfig:
+            birthMonth = birthdayConfig.get_attr(KEY_BDAY_MONTH)
+            birthDay = birthdayConfig.get_attr(KEY_BDAY_DAY)
+            if birthMonth and birthDay and await birthMonth() and await birthDay():
+                await ctx.send(
+                    f"{headerBad}\n"
+                    "Your birthday is already set. If you believe it is "
+                    "incorrect, please contact an admin or a moderator."
+                )
+                return
+
+        birthdayRoleId = await self.config.guild(ctx.guild).get_attr(KEY_BDAY_ROLE)()
+
+        if not birthdayRoleId:
+            await ctx.send(
+                f"{headerBad}\n"
+                "This server is not configured, please let a server "
+                "administrator or moderator know."
+            )
+            return
+
+        if not month and not day:
+            today = date.today()
+            month, day = today.month, today.day
+
+        try:
```

Comment:
Maybe add a confirmation dialogue to let the user check their input since they only have one chance?

Reply:
Sounds good.